ls1intum/Artemis
# `Lecture`: Fix issues with lecture attachments and quiz exercise re-evaluation

This PR fixes three issues:
1. When there is a quiz exercise as attachment unit in lecture, you cannot create additional `file` lecture units (e.g. PDFs) any more due to the serialization issue
`Could not write JSON: Cannot invoke \"java.lang.Boolean.booleanValue()\" because \"this.isVisibleBeforeStart\" is null'`
2. When there is a quiz exercise as attachment unit in a lecture, you cannot create additional attachments (directly in the lecture) due to the same serialization issue
3. When a quiz exercise includes categories, you cannot re-evaluate it any more

### Checklist
#### General



### Motivation and Context
I experienced the issues as instructor in my course

### Description
We remove some unnecessary data (which should also make the REST call faster) and convert categories correctly before re-evaluating quizzes

### Steps for Testing
1 Instructor
1 Student

1. Create a new lecture
2. Create a quiz (with 2-3 questions) with categories, connect it to a lecture (using lecture units) and let the quiz run
3. Re-evaluate the quiz --> should work
4. Add a new `file` lecture unit in the units section of the lecture (e.g. some PDF) --> should work without error
5. Add a new attachment directly to the lecture in the attachment section of the lecture (e.g. some PDF) --> should work without error
6. Display all units and attachments as student
7. Reorder the lecture units as instructor, update the attachments (in the unit and directly) with new file versions
8. Delete the attachments and lecture units as instructor

### Review Progress





#### Code Review
#### Manual Tests
